packages/backend/src/graphql/resolvers/incoming_payment.test.ts

```diff
@@ -68,4 +98,197 @@ describe('Incoming Payment Resolver', (): void => {
+    `(
+      '200 ($desc)',
+      async ({
+        description,
+        externalRef,
+        expiresAt,
+        incomingAmount
+      }): Promise<void> => {
+        const { id: paymentPointerId } = await createPaymentPointer(deps, {
+          asset
+        })
+        const payment = await createPayment({
+          paymentPointerId,
+          description,
+          externalRef,
+          expiresAt,
+          incomingAmount
+        })
+
+        const createSpy = jest
+          .spyOn(incomingPaymentService, 'create')
+          .mockResolvedValueOnce(payment)
+
+        const input = {
+          paymentPointerId: payment.paymentPointerId,
+          incomingAmount: payment.incomingAmount,
+          expiresAt: payment.expiresAt,
+          description,
+          externalRef
+        }
+
+        const query = await appContainer.apolloClient
+          .query({
+            query: gql`
+              mutation CreateIncomingPayment(
+                $input: CreateIncomingPaymentInput!
+              ) {
+                createIncomingPayment(input: $input) {
+                  code
+                  success
+                  message
+                  payment {
+                    id
+                    paymentPointerId
+                    state
+                    expiresAt
+                    incomingAmount {
+                      value
+                      assetCode
+                      assetScale
+                    }
+                    receivedAmount {
+                      value
+                      assetCode
+                      assetScale
+                    }
+                    description
+                    externalRef
+                    createdAt
+                  }
+                }
+              }
+            `,
+            variables: { input }
+          })
+          .then(
+            (query): IncomingPaymentResponse =>
+              query.data?.createIncomingPayment
+          )
+
+        expect(createSpy).toHaveBeenCalledWith(input)
+        expect(query).toEqual({
+          __typename: 'IncomingPaymentResponse',
+          code: '200',
+          success: true,
+          message: null,
+          payment: {
+            __typename: 'IncomingPayment',
+            id: payment.id,
+            paymentPointerId: payment.paymentPointerId,
+            state: SchemaPaymentState.Pending,
+            expiresAt: payment.expiresAt.toISOString(),
+            incomingAmount:
+              incomingAmount === undefined
+                ? null
+                : {
+                    __typename: 'Amount',
+                    ...serializeAmount(payment.incomingAmount)
+                  },
+            receivedAmount: {
+              __typename: 'Amount',
+              ...serializeAmount(payment.receivedAmount)
+            },
+            description: payment.description,
+            externalRef: payment.externalRef,
```

Comment:
While the values _should_ all be the same, I vote test against the input values vs the created payment
```suggestion
            paymentPointerId,
            state: SchemaPaymentState.Pending,
            expiresAt: expiresAt?.toISOString() || payment.expiresAt.toISOString(),
            incomingAmount:
              incomingAmount === undefined
                ? null
                : {
                    __typename: 'Amount',
                    ...serializeAmount(incomingAmount)
                  },
            receivedAmount: {
              __typename: 'Amount',
              ...serializeAmount(payment.receivedAmount)
            },
            description: description || null,
            externalRef: externalRef || null,
```